src/uct/ib/rc/base/rc_ep.h

```diff
@@ -353,7 +351,8 @@ uct_rc_txqp_completion_inl_resp(uct_rc_txqp_t *txqp, const void *resp, uint16_t
 static UCS_F_ALWAYS_INLINE uint8_t
 uct_rc_iface_tx_moderation(uct_rc_iface_t *iface, uct_rc_txqp_t *txqp, uint8_t flag)
 {
-    return (txqp->unsignaled >= iface->config.tx_moderation) ? flag : 0;
+    return ((txqp->unsignaled >= iface->config.tx_moderation) ||
```

Comment:
As i understand you are moving condition from this place to another one `uct_rc_txqp_posted` because we will need to update `iface->tx.real_moderation` on every `iface->tx.cq_available` change, i.e. on every post send according to changes  in `iface->tx.cq_available` (i mean reservation of CQ credit on every post for error case) + on completion